src/Features/CSharp/Portable/Completion/CompletionProviders/DeclarationNameCompletionProvider.cs

```diff
@@ -74,7 +74,7 @@ private ImmutableArray<ImmutableArray<string>> GetBaseNames(SemanticModel semant
                 return default;
             }
 
-            var (type, plural) = UnwrapType(nameInfo.Type, semanticModel.Compilation, wasPlural: false);
+            var (type, plural) = UnwrapType(nameInfo.Type, semanticModel.Compilation, wasPlural: false, seenTypes: new HashSet<ITypeSymbol>());
```

Comment:
:bulb: The majority of cases will get a performance boost if we lazily initialize this.

Reply:
Let's worry about that after we've shipped the minimal thing.